targets/TARGET_Cypress/TARGET_PSOC6/psoc6csp/abstraction/rtos/mbed_lib.json

```diff
@@ -0,0 +1,6 @@
+{
+    "name": "cy_pscoc6csp_rtos",
```

Comment:
Why is this needed? https://github.com/ARMmbed/mbed-os/tree/master/targets/TARGET_Cypress/TARGET_PSOC6/psoc6csp/abstraction/rtos/source/COMPONENT_RTX is included by targets.json (https://github.com/ARMmbed/mbed-os/blob/d78e009/targets/targets.json#L13425)

Reply:
The [baremetal configuration](https://os.mbed.com/docs/mbed-os/v5.15/reference/mbed-os-bare-metal.html) does not have an RTOS. This Mbed library file has been added to make all files in `targets/TARGET_Cypress/TARGET_PSOC6/psoc6csp/abstraction/rtos` part of a new library called `cy_pscoc6csp_rtos`. With this, I am able to use the macro `MBED_CONF_RTOS_PRESENT` to exclude code that calls RTOS functionality when building with the bare metal profile.